lua/telescope/_extensions/neoclip.lua

```diff
@@ -20,51 +22,74 @@ end
 local displayer = entry_display.create {
     separator = " ",
     items = {
-        { width = 65 },
+        { width = 60 },
         { remaining = true },
     },
 }
 
-local spec_per_type = {
+local spec_per_regtype = {
     c = 'charwise',
     l = 'linewise',
     b = 'blockwise',
 }
 
-local function make_display(entry)
-    local spec = spec_per_type[entry.type]
+local function spec_from_entry(entry)
+    local spec = spec_per_regtype[entry.regtype]
     local num_lines = #entry.contents
     if num_lines > 1 then
         spec = string.format('%s (%d lines)', spec, num_lines)
     end
-    return displayer {
-        entry.contents[1],
-        {spec, "Comment"},
+    return spec
+end
+
+local function make_display(entry)
+    local to_display = {entry.contents[1]}
+    if settings.content_spec_column then
+        table.insert(to_display, {spec_from_entry(entry), "Comment"})
+    end
+    return displayer(to_display)
+end
+
+local function entry_maker(entry)
+    return {
+        display = make_display,
+        contents = entry.contents,
+        regtype = entry.regtype,
+        filetype = entry.filetype,
+        ordinal = table.concat(entry.contents, '\n'),
+        -- TODO seem to be needed
+        name = 'name',
+        value = 'value', -- TODO what to put value to, affects sorting?
     }
 end
 
 local function get_export(register_name)
     return function(opts)
+        local previewer = false
+        if settings.preview then
+            previewer = previewers.new_buffer_previewer({
+                define_preview = function(self, entry, status)
+                    vim.api.nvim_buf_call(self.state.bufnr, function()
```

Comment:
Complete drive by comment, but I don't think you need the `nvim_buf_call` since you can just do 
`set_lines(bufnr)` and `vim.bo[bufnr].filetype = value`.

Reply:
Ah that makes sense! Thanks a lot for the comment @akinsho, really appreciate it :)

Reply:
Removed and indeed working without :+1: